embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/envelope/resource/DeviceImpl.kt

```diff
@@ -75,18 +75,9 @@ internal class DeviceImpl(
 
     private fun asyncRetrieveIsJailbroken() {
         // if the isJailbroken property exists in memory, don't try to retrieve it
-        if (isJailbroken != null) {
-            return
-        }
         backgroundWorker.submit {
-            val storedIsJailbroken = preferencesService.jailbroken
-            // load value from shared preferences
-            if (storedIsJailbroken != null) {
-                isJailbroken = storedIsJailbroken
-            } else {
-                isJailbroken = checkIfIsJailbroken()
-                preferencesService.jailbroken = isJailbroken
-            }
+            isJailbroken = checkIfIsJailbroken()
```

Comment:
do we want to always recompute, without checking if there's already a value in sharedPreferences?